packages/vuelidate/src/index.js

```diff
@@ -55,7 +55,7 @@ export function useVuelidate (validations, state, globalConfig = {}) {
     sendValidationResultsToParent,
     removeValidationResultsFromParent
   } = instance
-    ? nestedValidations({ $scope, instance })
+    ? nestedValidations({ $scope, instance: instance.proxy })
```

Comment:
You sure this is compatible with 2 2.7 and 3 ? I remember I had issues with the `getInstance` function a few times

Reply:
I made test with a real app only for 2.7, I trusted the units test for 2.6 and 3 but I'm going to try it

the function `getCurrentInstance` has different signature with 2.6 / 2.7 / 3

// 2.6
```js
declare function getCurrentInstance(): ComponentInternalInstance | null;
```

// 2.7
```js
export function getCurrentInstance(): { proxy: Vue } | null 
```

// 3
```js
export declare const getCurrentInstance: () => ComponentInternalInstance | null;
```

According to what has been said in this discussion:
https://github.com/vuejs/vue/issues/12596#issuecomment-1173269807

If I understand it right, we should NOT use the internal Instance but the public instance, which can be accessed by  `getCurrentInstance().proxy`

Maybe I'm wrong let me know what you understood about it

Reply:
Great find, let's keep it that way then. I mainly was not sure if vue 2 would have proxy, so was worried not to regress.

Reply:
I'm running Vue 3 and proxy doesn't have a `uid` on it. Means that the validations are registered as `_vuelidate_undefined`. Causing issues with multiple child validations.
```
$registerAs = `_vuelidate_${uid}`
```

Reply:
Sorry for that :( 

uid is not on the `ComponentPublicInstance` but on the  `ComponentInternalInstance` on vue3 

we can do as before by accessing the internal instance with `instance.$.uid`
```js
    const uid = instance?.$?.uid || instance._uid
    $registerAs = `_vuelidate_${uid}`
```

but as said before accessing internal instance is not recommended, 
maybe we should create our own uid ? I don't think we need to have the same as the component one.

any thoughts @dobromir-hristov ?